### DISCUSSION TITLE
Sending Command: Server returned 404: Not Found

### DISCUSSION THREAD

**Discussion by carlosafgraca** [llamafilm/tesla-http-proxy-addon#89] [General]
I can comunicate with TESLA API using Proxy like getting my public_key: "https://localhost:XXX/api/1/partner_accounts/public_key?domain=tesla.XYX.com" 

When I do some actions in "Tesla Custom Integration" using Tesla Proxy it works but, using custom HTTP request sometimes do not work. Someone can help me?

**set_sentry_mode on "Tesla Custom Integration" using Proxy:**
2024-05-02T10:19:29+01:00 [info ] Received POST request for /api/1/vehicles/VIN/command/set_sentry_mode
2024-05-02T10:19:29+01:00 [debug] Executing set_sentry_mode on VIN
2024-05-02T10:19:29+01:00 [info ] Starting dispatcher service...
2024-05-02T10:19:29+01:00 [info ] Session for DOMAIN_INFOTAINMENT loaded from cache
2024-05-02T10:19:29+01:00 [info ] Session for DOMAIN_VEHICLE_SECURITY loaded from cache
2024-05-02T10:19:29+01:00 [debug] Sending request to https://fleet-api.prd.eu.vn.cloud.tesla.com/api/1/vehicles/VIN/signed_command: {"routable_message":"MgIIAzoSBBBBBBBBBxh0DaQ6jvYE="}
2024-05-02T10:19:29+01:00 [debug] Server returned 200: OK: {"response":"MhISEArayk03viBBBBBBBBBBBEOoRnKLtFICCgA="}

**set_sentry_mode with HTTP Request using Proxy:**
2024-05-02T10:16:43+01:00 [info ] Received POST request for /api/1/vehicles/VIN/command/set_sentry_mode
2024-05-02T10:16:43+01:00 [debug] Executing set_sentry_mode on VIN
2024-05-02T10:16:43+01:00 [info ] Starting dispatcher service...
2024-05-02T10:16:43+01:00 [info ] Session for DOMAIN_INFOTAINMENT loaded from cache
2024-05-02T10:16:43+01:00 [info ] Session for DOMAIN_VEHICLE_SECURITY loaded from cache
2024-05-02T10:16:43+01:00 [debug] Sending request to https://fleet-api.prd.na.vn.cloud.tesla.com/api/1/vehicles/VIN/signed_command: {"routable_message":"MgIIAzAAAAAAAAAAAAAAAu6U+YlE="}
2024-05-02T10:16:43+01:00 [debug] Server returned 404: Not Found: {"response":null,"error":"not_found","error_description":"","txid":"12ac30aAAAAAAAAAAcb0ef3c26"}

**Sample HTTP Query:**
var token = global.get("TESLA_TOKEN");
msg.url = "https://localhost:4430/api/1/vehicles/VIN/command/set_sentry_mode";
msg.headers = {
    "Authorization": "Bearer " + token,
    "Content-Type": "application/json"
};
msg.payload = {
    "on": true
};
msg.method="POST";

**Comment by llamafilm** [comment]
You said this happens “sometimes”. Any ideas of what the pattern is? Does it always fail with sentry, or what about other commands?

Guessing:  You could look through the integration’s source code to see what headers it includes; maybe you’re missing one.

**Comment by carlosafgraca** [comment]
I'm sorry for my bad explaination. 

If I send a bad constructed JSON payload to "ON" command like "ONN":true the API return that can not find "ON" command but, if I send "ON": true it send me 404 error. Seems like I can comunicate and the command index are intrepreted well but, it falls in 404 error. I get that error im all veichle commands. The command value (true in case) should be encrypted in HTTP request to Tesla HTTP proxy?

**Missing "ON" command Sample HTTP Query:**
var token = global.get("TESLA_TOKEN");
msg.url = "https://localhost:4430/api/1/vehicles/VIN/command/set_sentry_mode";
msg.headers = {
"Authorization": "Bearer " + token,
"Content-Type": "application/json"
};
msg.payload = {
**"onn"**: true
};
msg.method="POST";


**404 Sample HTTP Query:**
var token = global.get("TESLA_TOKEN");
msg.url = "https://localhost:4430/api/1/vehicles/VIN/command/set_sentry_mode";
msg.headers = {
"Authorization": "Bearer " + token,
"Content-Type": "application/json"
};
msg.payload = {
**"on"**: true
};
msg.method="POST";


I found that issue only in all command vechicle mensages so, I can get 200 OK and a good output with commands like bellow:

**OK Command 1:**
var token = global.get("TESLA_TOKEN");
msg.url = "https://localhost:4430/api/1/users/feature_config";
msg.headers = {
    "Authorization": "Bearer " + token,
    "Content-Type": "application/json"
};
msg.payload = {};
msg.method="GET";
return msg;

**OUTPUT:**
{"response":{"signaling":{"enabled":true,"subscribe_connectivity":false,"use_auth_token":false},"auth_rejection":{"enabled":true,"min_version":4}}}

var token = global.get("TESLA_TOKEN");

**OK Command 2:**
msg.url = "https://localhost:4430/api/1/partner_accounts/public_key?domain=tesla.XXX.com";
msg.headers = {
    "Authorization": "Bearer " + token,
    "Content-Type": "application/json"
};
msg.payload = {};
msg.method="GET";
return msg;

**OUTPUT:**
{"response":{"public_key":"04fc66e43f777883AAABBBCCCf9ffd6d5b175"}}

I guess that communication are good but I had some problem in signed command. With that info can you help me?

**Comment by llamafilm** [comment]
Have you tried testing with the curl example from the documentation? https://developer.tesla.com/docs/fleet-api

I wonder if your “true” Boolean needs to be quoted or capitalized.

But if the curl example doesn’t work then I suggest raising this on the vehicle-command repo.

**Comment by carlosafgraca** [comment]
Thank you llamafilm! As sugested I run 4 tests diretly with CURL using diferent input data:

**TEST 1:** --data '{     "on": "true" }'
**TEST 2:** --data '{     "on": true   }'
**TEST 3:** --data '{     "on": "TRUE" }'
**TEST 4:** --data '{     "on": TRUE   }'

I already do the test using TESLA HTTP Proxy with the same input data and the results are the same. Can you please take a look?

**TEST 1**
	**_INPUT:_**
	curl --header 'Content-Type: application/json' \   --header "Authorization: Bearer AAA.BBB.CCC" \   **--data '{     "on": "true"   }'** \   'https://fleet-api.prd.eu.vn.cloud.tesla.com/api/1/vehicles/VIN/command/set_sentry_mode' 
	**_Output:_**
		{"_msgid":"f5966535c4","topic":"","payload":"{\"response\":null,\"error\":\"not_found\",\"error_description\":\"\",\"txid\":\"3bb58f91aa9353fa\"}","rc":{"code":0}}
	**_RESPONSE:_**	
		{"response":null,"error":"not_found","error_description":"","txid":"3bb32301aa9353fa"}
	**_ERROR:_** 	
		{"_msgid":"f596655c4","topic":"","payload":"curl: (3) URL rejected: Malformed input to a URL function\ncurl: (3) URL rejected: Malformed input to a URL function\ncurl: (3) URL rejected: Malformed input to a URL function\n  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r100   124  100   102  100    22    150     32 --:--:-- --:--:-- --:--:--   183\n","rc":{"code":0}}

**TEST 2**
	**_INPUT:_** 
	curl --header 'Content-Type: application/json' \   --header "Authorization: Bearer AAA.BBB.CCC" \   **--data '{     "on": true   }'** \   'https://fleet-api.prd.eu.vn.cloud.tesla.com/api/1/vehicles/VIN/command/set_sentry_mode' 
	**_Output:_**
		{"_msgid":"e14dcb5543","topic":"","payload":"{\"response\":null,\"error\":\"not_found\",\"error_description\":\"\",\"txid\":\"1593a503ea29a1\"}","rc":{"code":0}}
	**_RESPONSE:_**	
		{"response":null,"error":"not_found","error_description":"","txid":"1593a50fe3ea29a1"}
	**_ERROR:_** 	
		{"_msgid":"e14dcb33601b5543","topic":"","payload":"curl: (3) URL rejected: Malformed input to a URL function\ncurl: (3) URL rejected: Malformed input to a URL function\ncurl: (3) URL rejected: Malformed input to a URL function\n  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r100    20    0     0  100    20      0     34 --:--:-- --:--:-- --:--:--    34\r100   122  100   102  100    20    160     31 --:--:-- --:--:-- --:--:--   192\n","rc":{"code":0}}


**TEST 3**
	**_INPUT:_** 
	curl --header 'Content-Type: application/json' \   --header "Authorization: Bearer AAA.BBB.CCC" \   **--data '{     "on": "TRUE"   }'** \   'https://fleet-api.prd.eu.vn.cloud.tesla.com/api/1/vehicles/VIN/command/set_sentry_mode' 
	**_Output:_**
		{"_msgid":"8ae60722f","topic":"","payload":"{\"response\":null,\"error\":\"not_found\",\"error_description\":\"\",\"txid\":\"9d21da0439ad1320\"}","rc":{"code":0}}
	**_RESPONSE:_**
		{"response":null,"error":"not_found","error_description":"","txid":"9d21da047d1320"}
	**_ERROR:_** 	
		{"_msgid":"8ae608892a85722f","topic":"","payload":"curl: (3) URL rejected: Malformed input to a URL function\ncurl: (3) URL rejected: Malformed input to a URL function\ncurl: (3) URL rejected: Malformed input to a URL function\n  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r100   124  100   102  100    22    159     34 --:--:-- --:--:-- --:--:--   194\n","rc":{"code":0}}


**TEST 4**
	**_INPUT:_** 
	curl --header 'Content-Type: application/json' \   --header "Authorization: Bearer AAA.BBB.CCC" \   **--data '{     "on": TRUE   }'** \   'https://fleet-api.prd.eu.vn.cloud.tesla.com/api/1/vehicles/VIN/command/set_sentry_mode' 
	**_Output:_**
		{"_msgid":"f7c9b6afd1354a3d","topic":"","payload":"{\"error\":\"invalid JSON: Error occurred while parsing request parameters\"}","rc":{"code":0}}
	**_RESPONSE:_**
		{"error":"invalid JSON: Error occurred while parsing request parameters"}
	**_ERROR:_** 	
		{"_msgid":"f7c9b6afd1354a3d","topic":"","payload":"curl: (3) URL rejected: Malformed input to a URL function\ncurl: (3) URL rejected: Malformed input to a URL function\ncurl: (3) URL rejected: Malformed input to a URL function\n  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r100    93    0    73  100    20    113     31 --:--:-- --:--:-- --:--:--   145\n","rc":{"code":0}}


**USED Authorization: Bearer TOKEN:**
{
  "gty": "client-credentials",
  "sub": "a2XXXXXb7",
  "iss": "https://auth.tesla.com/oauth2/v3/nts",
  "azp": "d2YYYYY9a",
  "aud": [
    "https://auth.tesla.com/oauth2/v3/clientinfo",
    "https://fleet-api.prd.eu.vn.cloud.tesla.com"
  ],
  "exp": 1715024437,
  "iat": 1714995637,
  "scp": [
    "user_data",
    "vehicle_device_data",
    "vehicle_cmds",
    "vehicle_charging_cmds",
    "energy_device_data",
    "energy_cmds",
    "openid",
    "offline_access"
  ]
}
